src/System.Runtime/tests/System/Array.cs

```diff
@@ -297,6 +298,73 @@ public static void TestBinarySearch()
     }
 
     [Fact]
+    public static void TestBinarySearch_NoIComparer()
+    {
+        int idx;
+        int[] ia = { 1, 3, 6, 6, 8, 10, 12, 16 };
+        idx = Array.BinarySearch((Array)ia, 8);
+        Assert.Equal(idx, 4);
+
+        //Return value semantics: Quoted from MSDN without comment.
+        //
+        //The index of the specified value in the specified array, 
+        //if value is found. If value is not found and value is 
+        //less than one or more elements in array, a negative 
+        //number which is the bitwise complement of the index of 
+        //the first element that is larger than value. 
+        //If value is not found and value is greater than any 
+        //of the elements in array, a negative number which is the bitwise complement of (the index of the last element plus 1).
```

Comment:
This was copied from an older test. I have fixed this now by rewriting the test as a xunit theory.